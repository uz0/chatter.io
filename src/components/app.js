import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Router, Route, browserHistory, IndexRedirect } from 'react-router';

import Layout from '@/components/layout';

import SignIn from '@/components/sign-in_container';
import SignUp from '@/components/sign-up_container';
import ForgotPassword from '@/components/forgot-password_container';
import Chat from '@/components/chat_container';

import store from '@/store';

export default class App extends Component {
  render() {
    return <Provider store={store}>
      <Router history={browserHistory}>
        <Route path="/" component={Layout}>
          <IndexRedirect to="chat" />
          <Route component={Chat} path="chat" />
          <Route component={SignIn} path="sign-in" />
          <Route component={SignIn} path="sign-in/:code" />
          <Route component={SignUp} path="sign-up" />
          <Route component={SignUp} path="sign-up/:code" />
          <Route component={ForgotPassword} path="/forgot-password" />
        </Route>
      </Router>
    </Provider>;
  }
}
