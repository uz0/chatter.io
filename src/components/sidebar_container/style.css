.sidebar {
  --sidebar-padding: 12px;
  --divider-color: #f3f4f7;

  position: relative;
  display: flex;
  flex-direction: column;
  max-width: 420px;
}

.sidebar .header {
  flex-shrink: 0;
  display: flex;
  align-items: flex-end;
  height: var(--info-height);
  padding-left: var(--sidebar-padding);
  padding-right: var(--sidebar-padding);
}

.sidebar .header h1 {
  --height: 38px;
  overflow: hidden;
  flex-grow: 1;
  min-width: 0;
  height: var(--height);
  line-height: var(--height);
  font-size: 15px;
  font-weight: var(--font-weight-semibold);
  white-space: nowrap;
  text-overflow: ellipsis;
}

.sidebar .header .dropdown {
  --size: 38px;
  flex-shrink: 0;
  margin-left: 12px;
  height: var(--size);
}

.sidebar .header .dropdown > button {
  width: var(--size);
  height: var(--size);
  background-image: var(--bg-image);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  border-radius: calc(var(--size) / 2);
}

.sidebar .header .button {
  --height: 38px;
  --icon-size: 18px;
  flex-shrink: 0;
  margin-left: 10px;
}

.sidebar .search {
  flex-shrink: 0;
  margin-left: var(--sidebar-padding);
  margin-right: var(--sidebar-padding);
}

.sidebar .navigation {
  flex-shrink: 0;
  display: flex;
  margin-top: 16px;
  margin-left: var(--sidebar-padding);
  margin-right: var(--sidebar-padding);
  background-color: var(--divider-color);
  border-radius: 5px;
}

.sidebar .navigation button {
  --text-color: #9ca0b0;
  flex-grow: 1;
  flex-basis: 0;
  position: relative;
  height: 38px;
  font-size: 15px;
  text-align: center;
  font-weight: var(--font-weight);
  color: var(--text-color);
  background-color: var(--bg-color);
  transition-property: color;
}

.sidebar .navigation button:first-of-type:not(._is-active) {
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.sidebar .navigation button:last-of-type:not(._is-active) {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.sidebar .navigation button:not(._is-active):not(:hover) {
  --font-weight: var(--font-weight-medium);
  --bg-color: var(--divider-color);
}

.sidebar .navigation button:not(._is-active):hover {
  --font-weight: var(--font-weight-medium);
  --text-color: #000;
}

.sidebar .navigation button._is-active {
  --font-weight: var(--font-weight-bold);
  --text-color: #252c43;
  --bg-color: #fff;

  border-radius: 5px;
  box-shadow: inset 0 0 0 1px var(--divider-color);
}

.sidebar .list {
  position: relative;
  overflow-x: hidden;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  min-height: 0;
  -webkit-overflow-scrolling: touch;
}