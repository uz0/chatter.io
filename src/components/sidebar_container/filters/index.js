import React, { Component, Fragment } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import get from 'lodash/get';
import uniq from 'lodash/uniq';
import isEmpty from 'lodash/isEmpty';
import Section from '@/components/sidebar_container/section';
import SubscriptionItem from '@/components/subscription-item';
import SubscriptionAvatar from '@/components/subscription-avatar';
import { api } from '@';
import { withSortedSubscriptions, withRouter } from '@/hoc';
import { getChatName, uid, getOpponentUser } from '@/helpers';
import { actions as subscriptionsActions } from '@/store/subscriptions';
import { actions as messagesActions } from '@/store/messages';
import { actions as usersActions } from '@/store/users';
import style from './style.css';

class Filters extends Component {
  getSubscriptionHref = subscription => {
    if (subscription.group.type === 'private_chat' && !isEmpty(getOpponentUser(subscription))) {
      return `/chat/user/${getOpponentUser(subscription).id}`;
    }

    return `/chat/${subscription.id}`;
  };

  stopSearchingMessages = () => this.props.toggleSearchMessages(false);

  goToMessage = params => () => {
    const subscription = this.props.subscriptions_list[params.chatId];
    const href = this.getSubscriptionHref(subscription);
    this.props.pushUrl(`${href}/${params.messageId}`);
    const message = this.props.messages_list[params.messageId];

    let text = message.text;

    uniq(message.text.split(' ')).forEach(word => {
      if (!word.match(this.props.subscriptions_filter_text)) {
        return;
      }

      const regex = new RegExp(word, 'g');
      text = text.replace(regex, `<span>${word}</span>`);
    });

    this.props.updateMessage({
      chatId: params.chatId,

      message: {
        ...message,
        text,
      },
    });

    setTimeout(() => {
      this.props.updateMessage({
        chatId: params.chatId,

        message: {
          ...message,
          text: message.text,
        },
      });
    }, 3000);
  };

  goToChat = id => () => {
    const subscription = this.props.subscriptions_list[id];
    const href = this.getSubscriptionHref(subscription);
    this.props.pushUrl(href);
  };

  goToGlobalChat = id => () => {
    api.getPrivateSubscription({ user_id: id }).then(data => {
      if (this.props.subscriptions_ids.indexOf(data.subscription.id) === -1) {
        this.props.addUsers(data.subscription.group.participants);
        this.props.addSubscription(data.subscription);
      }

      this.props.pushUrl(`/chat/user/${id}`);
    });
  };

  getFilteredMessages = sortedSubscriptionsIds => {
    if (!sortedSubscriptionsIds) {
      return [];
    }

    let messages = [];

    this.props.sorted_subscriptions_ids.forEach(chatId => {
      const chatMessages = this.props.subscriptions_filtered_messages[chatId] || [];

      chatMessages.forEach(messageId => {
        messages.push({ chatId, messageId });
      });
    });

    return messages;
  };

  renderContact = ({ item }) => {
    const subscription = this.props.subscriptions_list[item];
    const name = getChatName(subscription);

    return <div
      key={item}
      className={style.contact}
      onClick={this.goToChat(subscription.id)}
    >
      <SubscriptionAvatar subscription={subscription} className={style.avatar} />
      <p className={style.name}>{name}</p>
    </div>;
  };

  renderGlobalContact = ({ item }) => {
    const name = get(item, 'nick', 'no nick') || 'no nick';

    return <div
      key={item.id}
      className={style.contact}
      onClick={this.goToGlobalChat(item.id)}
    >
      <SubscriptionAvatar user={item} className={style.avatar} />
      <p className={style.name}>{name}</p>
    </div>;
  };

  renderSubscription = ({ item }) => {
    return <SubscriptionItem
      key={uid()}
      id={item.chatId}
      messageId={item.messageId}
      onClick={this.goToMessage({ chatId: item.chatId, messageId: item.messageId })}
      className={style.subscription}
    />;
  };

  render() {
    const filteredMessages = this.getFilteredMessages(this.props.sorted_subscriptions_ids);
    const stopSearchAction = { text: this.props.t('stop_loading'), onClick: this.stopSearchingMessages};

    return <Fragment>
      <Section
        title={this.props.t('contact_plural')}
        emptyMessage={this.props.t('no_results')}
        items={this.props.subscriptions_filtered_contacts_ids}
        renderItem={this.renderContact}
      />

      <Section
        title={this.props.t('global_search')}
        emptyMessage={this.props.t('no_results')}
        items={this.props.subscriptions_filtered_global_users}
        renderItem={this.renderGlobalContact}
      />

      <Section
        title={this.props.t('message_plural')}
        isLoading={this.props.is_searching_old_messages}
        emptyMessage={this.props.t('no_results')}
        items={filteredMessages}
        renderItem={this.renderSubscription}
        {...this.props.is_searching_old_messages ? {action: stopSearchAction} : {}}
      />
    </Fragment>;
  }
}

export default compose(
  withRouter,
  withTranslation(),

  connect(
    state => ({
      subscriptions_ids: state.subscriptions.ids,
      subscriptions_list: state.subscriptions.list,
      subscriptions_filtered_messages: state.subscriptions.filtered_messages,
      subscriptions_filtered_contacts_ids: state.subscriptions.filtered_contacts_ids,
      subscriptions_filtered_global_users: state.subscriptions.filtered_global_users,
      subscriptions_filtered_ids: state.subscriptions.filtered_ids,
      subscriptions_filter_text: state.subscriptions.filter_text,
      is_searching_old_messages: state.subscriptions.is_searching_old_messages,
      messages_list: state.messages.list,
    }),

    {
      addSubscription: subscriptionsActions.addSubscription,
      toggleSearchMessages: subscriptionsActions.toggleSearchMessages,
      updateMessage: messagesActions.updateMessage,
      addUsers: usersActions.addUsers,
    },
  ),

  withSortedSubscriptions(props => ({
    ids: props.subscriptions_filtered_ids,
  })),
)(Filters);
