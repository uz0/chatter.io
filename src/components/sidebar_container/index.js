import React, { Component } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';
import { withNamespaces } from 'react-i18next';
import classnames from 'classnames/bind';
import SubscriptionItem from '@/components/subscription-item';
import Button from '@/components/button';
import SearchInput from '@/components/search-input';
import { api } from '@';
import { actions as subscriptionsActions } from '@/store/subscriptions';
import { actions as modalActions } from '@/components/modal_container';
import style from './style.css';

const cx = classnames.bind(style);

class Sidebar extends Component {
  state = {
    navigationActive: 'all',
  };

  chooseTabNavigation = tab => () => this.setState({ navigationActive: tab });

  openAddChat = () => this.props.showModal('new-chat-modal');

  async componentWillMount() {
    const shortSubscriptions = await api.getSubscriptions({ short: true });
    this.props.loadSubscriptionsIds(shortSubscriptions.subscriptions);
  }

  render() {
    return <div className={cx('sidebar', this.props.className)}>
      <div className={style.header}>
        <h1>Unichat</h1>
        <div className={style.image} style={{ '--bg-image': 'url(/assets/default-user.jpg)' }} />
        <Button appearance="_fab-divider" icon="add-chat" onClick={this.openAddChat} className={style.button} />
      </div>

      <SearchInput className={style.search} />

      <div className={style.navigation}>
        <button
          className={cx({'_is-active': this.state.navigationActive === 'all'})}
          onClick={this.chooseTabNavigation('all')}
        >All</button>

        <button
          className={cx({'_is-active': this.state.navigationActive === 'personal'})}
          onClick={this.chooseTabNavigation('personal')}
        >Personal</button>

        <button
          className={cx({'_is-active': this.state.navigationActive === 'work'})}
          onClick={this.chooseTabNavigation('work')}
        >Work</button>
      </div>

      <div className={style.list}>
        {this.props.subscriptions_ids &&
          this.props.subscriptions_ids.map(id => <SubscriptionItem
            key={id}
            id={id}
            className={style.subscription}
          />)}

        {!this.props.subscriptions_ids &&
          <p className={style.empty}>There is no chats</p>}
      </div>
    </div>;
  }
}

export default compose(
  withNamespaces('translation'),

  connect(
    state => ({
      subscriptions_ids: state.subscriptions.ids,
      subscriptions_list: state.subscriptions.list,
    }),

    {
      loadSubscriptionsIds: subscriptionsActions.loadSubscriptionsIds,
      showModal: modalActions.showModal,
    },
  ),
)(Sidebar);
