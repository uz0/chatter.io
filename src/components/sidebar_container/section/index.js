import React from 'react';
import classnames from 'classnames/bind';
import Loading from '@/components/loading';
import style from './style.css';

const cx = classnames.bind(style);

const Section = ({ items, isLoading, action, title, emptyMessage, renderItem, className }) => {
  const isItemsExist = items && items.length > 0;

  return <div className={cx('section', className)}>
    {title &&
      <div className={style.title}>
        <p className={style.name}>{title}</p>

        {isLoading &&
          <Loading isShown className={style.loading} />
        }

        {action &&
          <button className={style.action} onClick={action.onClick}>{action.text}</button>
        }
      </div>

    }

    {isItemsExist &&
      items.map(item => renderItem({ item }))
    }

    {!isItemsExist &&
      <p className={style.empty}>{emptyMessage}</p>
    }
  </div>;
};

export default Section;
