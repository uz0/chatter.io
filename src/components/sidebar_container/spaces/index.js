import React, { Component } from 'react';
import style from './style.css';

class Spaces extends Component {
  render() {
    return <p className={style.spaces}>Spaces</p>;
  }
}

export default Spaces;
