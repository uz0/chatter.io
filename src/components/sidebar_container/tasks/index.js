import React, { Component, Fragment } from 'react';
import compose from 'recompose/compose';
import get from 'lodash/get';
import groupBy from 'lodash/groupBy';
import filter from 'lodash/filter';
import { connect } from 'react-redux';
import Icon from '@/components/icon';
import Task from '@/components/task';
import SubscriptionAvatar from '@/components/subscription-avatar';
import { ClickOutside } from 'reactjs-click-outside';
import { api } from '@';
import { actions as modalActions } from '@/components/modal_container';
import { actions as tasksActions } from '@/store/tasks';
import classnames from 'classnames/bind';
import style from './style.css';

const cx = classnames.bind(style);

class Tasks extends Component {
  state = {
    filter: 'all',
  };

  setTab = filter => () => this.setState({ filter });

  newTodo = () => this.props.toggleModal({
    id: 'classic-edit-task-modal',

    options: {
      organization_id: this.props.organization_id,
    },
  });

  openTask = id => () => this.props.toggleModal({
    id: 'classic-edit-task-modal',

    options: {
      task_id: id,
    },
  });

  async componentWillMount() {
    if (!this.props.isLoaded) {
      const { tasks } = await api.organizationTasks({ organization_id: this.props.organization_id });
      this.props.loadOrganizationTasks(tasks);
    }
  }

  async componentWillReceiveProps(nextProps) {
    const isOrganizationChanged = this.props.organization_id !== nextProps.organization_id;

    if (isOrganizationChanged && !nextProps.isLoaded) {
      const { tasks } = await api.organizationTasks({ organization_id: nextProps.organization_id });
      this.props.loadOrganizationTasks(tasks);
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    const isOrganizationChanged = this.props.organization_id !== nextProps.organization_id;
    const isCountTasksFromMeChanged = this.props.tasks.length !== nextProps.tasks.length;
    const ifIsLoadedChanged = this.props.isLoaded !== nextProps.isLoaded;
    const isFilterChanged = this.state.filter !== nextState.filter;

    return isOrganizationChanged ||
      ifIsLoadedChanged ||
      isFilterChanged ||
      isCountTasksFromMeChanged;
  }

  render() {
    let tasks = filter(this.props.tasks, task => !!task.group_id);

    if (this.state.filter === 'my') {
      tasks = filter(tasks, {executor_id: this.props.currentUserId});
    }

    const groupedTasks = groupBy(tasks, 'group_id');
    const isTasksExist = Object.keys(groupedTasks).length > 0;

    return <ClickOutside outsideCall={this.props.close}>
      <div className={cx('popup', this.props.className)}>
        <div className={style.navigation}>
          <button
            type="button"
            className={cx('tab', {'_is-active': this.state.filter === 'all'})}
            onClick={this.setTab('all')}
          >All tasks</button>

          <button
            type="button"
            className={cx('tab', {'_is-active': this.state.filter === 'my'})}
            onClick={this.setTab('my')}
          >My tasks</button>

          <button className={style.open}>
            <Icon name="full-screen-half" />
          </button>
        </div>

        <div className={style.list}>
          <button className={style.new} onClick={this.newTodo}>+ New To-Do</button>

          {Object.keys(groupedTasks).map(key => {
            const group = groupedTasks[key];
            const groupId = group[0].group_id;

            return <Fragment key={key}>
              <div className={style.section}>
                <SubscriptionAvatar className={style.avatar} groupId={groupId} />
                <p className={style.name}>123</p>
                <button className={style.all}>See all</button>
              </div>

              {group.map(task => {
                return <Task
                  key={task.id}
                  id={task.id}
                  onClick={this.openTask(task.id)}
                  className={style.task}
                />;
              })}
            </Fragment>;
          })}

          {!isTasksExist &&
            <p className={style.empty}>There is no tasks</p>
          }
        </div>
      </div>
    </ClickOutside>;
  }
}

export default compose(
  connect(
    (state, props) => ({
      currentUserId: state.currentUser.id,
      isLoaded: get(state.tasks.organizations, `${props.organization_id}.isLoaded`, false),
      tasks: filter(state.tasks.list, {organization_id: props.organization_id}),
    }),

    {
      loadOrganizationTasks: tasksActions.loadOrganizationTasks,
      toggleModal: modalActions.toggleModal,
    },
  ),
)(Tasks);
