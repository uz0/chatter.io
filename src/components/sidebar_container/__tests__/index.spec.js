import React from 'react';
import { shallow } from 'enzyme';
import { Sidebar } from '@/components/sidebar_container';

describe('Sidebar component', () => {
  it('renders correctly', () => {
    const currentUser = {
      auth_token: 'Qli089GxuOrWGAAVYzcVJH1px_lRHO6Wf0Ke_icqf8kpndF0Fz_b02li8',

      avatar: {
        full: 'https://chat.mainnetwork.io/rails/active_storage/b…9eb29647350acdf569189259ae78b83/inline_image.jpeg',
        small: 'https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg',
      },

      confirmed_at: '2018-10-16T12:06:32Z',
      email: 'efim1382@gmail.com',
      id: 55,
      is_confirmed: true,
      nick: 'efim1382',
      searchable_nick: true,
    };

    const wrapper = shallow(<Sidebar
      t={() => {}}
      subscriptions_ids={[]}
      sorted_subscriptions_ids={[]}
      currentUser={currentUser}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
