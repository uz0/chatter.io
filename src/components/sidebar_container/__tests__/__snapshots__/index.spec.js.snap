// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    currentUser={
      Object {
        "auth_token": "Qli089GxuOrWGAAVYzcVJH1px_lRHO6Wf0Ke_icqf8kpndF0Fz_b02li8",
        "avatar": Object {
          "full": "https://chat.mainnetwork.io/rails/active_storage/b…9eb29647350acdf569189259ae78b83/inline_image.jpeg",
          "small": "https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg",
        },
        "confirmed_at": "2018-10-16T12:06:32Z",
        "email": "efim1382@gmail.com",
        "id": 55,
        "is_confirmed": true,
        "nick": "efim1382",
        "searchable_nick": true,
      }
    }
    sorted_subscriptions_ids={Array []}
    subscriptions_ids={Array []}
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <h1>
            Unichat
          </h1>
          <Connect(Dropdown)
            className="dropdown"
            items={
              Array [
                Object {
                  "onClick": [Function],
                  "text": undefined,
                },
                Object {
                  "onClick": [Function],
                  "text": undefined,
                },
              ]
            }
            uniqueId="sidebar-user-dropdown"
          >
            <button
              className="image"
              style={
                Object {
                  "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                }
              }
            />
          </Connect(Dropdown)>
          <Button
            appearance="_fab-divider"
            className="button"
            icon="add-chat"
            onClick={[Function]}
          />
        </div>,
        <LoadNamespace(SearchInput)
          className="search"
          onInput={[Function]}
        />,
        <div
          className="navigation"
        >
          <button
            className=""
            onClick={[Function]}
          />
          <button
            className=""
            onClick={[Function]}
          />
          <button
            className=""
            onClick={[Function]}
          />
        </div>,
        undefined,
        <div
          className="list _is-loading"
        />,
      ],
      "className": "sidebar",
      "data-c5c61f8c": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Unichat
            </h1>,
            <Connect(Dropdown)
              className="dropdown"
              items={
                Array [
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                ]
              }
              uniqueId="sidebar-user-dropdown"
            >
              <button
                className="image"
                style={
                  Object {
                    "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                  }
                }
              />
            </Connect(Dropdown)>,
            <Button
              appearance="_fab-divider"
              className="button"
              icon="add-chat"
              onClick={[Function]}
            />,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Unichat",
            },
            "ref": null,
            "rendered": "Unichat",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="image"
                style={
                  Object {
                    "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                  }
                }
              />,
              "className": "dropdown",
              "items": Array [
                Object {
                  "onClick": [Function],
                  "text": undefined,
                },
                Object {
                  "onClick": [Function],
                  "text": undefined,
                },
              ],
              "uniqueId": "sidebar-user-dropdown",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "image",
                "style": Object {
                  "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appearance": "_fab-divider",
              "className": "button",
              "icon": "add-chat",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "search",
          "onInput": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className=""
              onClick={[Function]}
            />,
            <button
              className=""
              onClick={[Function]}
            />,
            <button
              className=""
              onClick={[Function]}
            />,
          ],
          "className": "navigation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "list _is-loading",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <h1>
              Unichat
            </h1>
            <Connect(Dropdown)
              className="dropdown"
              items={
                Array [
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                ]
              }
              uniqueId="sidebar-user-dropdown"
            >
              <button
                className="image"
                style={
                  Object {
                    "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                  }
                }
              />
            </Connect(Dropdown)>
            <Button
              appearance="_fab-divider"
              className="button"
              icon="add-chat"
              onClick={[Function]}
            />
          </div>,
          <LoadNamespace(SearchInput)
            className="search"
            onInput={[Function]}
          />,
          <div
            className="navigation"
          >
            <button
              className=""
              onClick={[Function]}
            />
            <button
              className=""
              onClick={[Function]}
            />
            <button
              className=""
              onClick={[Function]}
            />
          </div>,
          undefined,
          <div
            className="list _is-loading"
          />,
        ],
        "className": "sidebar",
        "data-c5c61f8c": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Unichat
              </h1>,
              <Connect(Dropdown)
                className="dropdown"
                items={
                  Array [
                    Object {
                      "onClick": [Function],
                      "text": undefined,
                    },
                    Object {
                      "onClick": [Function],
                      "text": undefined,
                    },
                  ]
                }
                uniqueId="sidebar-user-dropdown"
              >
                <button
                  className="image"
                  style={
                    Object {
                      "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                    }
                  }
                />
              </Connect(Dropdown)>,
              <Button
                appearance="_fab-divider"
                className="button"
                icon="add-chat"
                onClick={[Function]}
              />,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Unichat",
              },
              "ref": null,
              "rendered": "Unichat",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="image"
                  style={
                    Object {
                      "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                    }
                  }
                />,
                "className": "dropdown",
                "items": Array [
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                  Object {
                    "onClick": [Function],
                    "text": undefined,
                  },
                ],
                "uniqueId": "sidebar-user-dropdown",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "image",
                  "style": Object {
                    "--bg-image": "url(https://chat.mainnetwork.io/rails/active_storage/r…e8e3e554111dab5bd87656de36a7063/inline_image.jpeg)",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "appearance": "_fab-divider",
                "className": "button",
                "icon": "add-chat",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "search",
            "onInput": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className=""
                onClick={[Function]}
              />,
              <button
                className=""
                onClick={[Function]}
              />,
              <button
                className=""
                onClick={[Function]}
              />,
            ],
            "className": "navigation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "list _is-loading",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
